pkg/controller/machineset/delete_policy.go

```diff
@@ -67,6 +76,34 @@ func oldestDeletePriority(machine *v1alpha1.Machine) deletePriority {
 	return deletePriority(float64(mustDelete) * (1.0 - math.Exp(-d.Seconds()/secondsPerTenDays)))
 }
 
+
+func customDeletePolicy(machine *v1alpha1.Machine) deletePriority {
+	// Iterate through conditions to get if machine is not "Ready"
+	// Check types.go
+	for _, c := range machine.Status.Conditions {
+		if c.Type != corev1.NodeReady {
+			return mustDelete
+		}
+	}
+	// Rest of conditions are extracted from the "Newest" policy, that is the one that
+	// matches the idea of deleting the newest machines (more possibilities to be WIP in the attachment to cluster)
+	// than the old ones.
+	if machine.DeletionTimestamp != nil && !machine.DeletionTimestamp.IsZero() {
+		return mustDelete
+	}
+	// If annotations are set + DeleteAnnotation is present
+	if machine.ObjectMeta.Annotations != nil && machine.ObjectMeta.Annotations[DeleteNodeAnnotation] != "" {
```

Comment:
Unnecessary nil check:
```suggestion
	if v, ok := machine.ObjectMeta.Annotations[DeleteNodeAnnotation]; ok && v != "" {
```